Source/CarthageKit/Project.swift

```diff
@@ -728,6 +756,47 @@ private func dSYMForFramework(frameworkURL: NSURL, inDirectoryURL directoryURL:
 		|> take(1)
 }
 
+/// Sends the URL to each bcsymbolmap found in the given directory.
+private func BCSymbolMapsInDirectory(directoryURL: NSURL) -> SignalProducer<NSURL, CarthageError> {
+	return filesInDirectory(directoryURL, fileExtension: "bcsymbolmap")
+}
+
+/// Sends the URLs of the bcsymbolmap files that match the given framework and are
+/// located somewhere within the given directory.
+private func BCSymbolMapsForFramework(frameworkURL: NSURL, inDirectoryURL directoryURL: NSURL) -> SignalProducer<NSURL, CarthageError> {
+	return UUIDsForFramework(frameworkURL)
+		|> flatMap(.Merge) { UUIDs in
+			if UUIDs.isEmpty {
+				return SignalProducer.empty
+			}
+			// conceptually, we just want to filter the URLs to find any that are in uuids.
+			// but we'd like to stop enumeration early when we find all the files.
+			// this also guarantees we don't send two URLs for bcsymbolmap files with the same UUID.
+			return SignalProducer { observer, disposable in
+				var remainingUUIDs = UUIDs
+				BCSymbolMapsInDirectory(directoryURL)
+					|> startWithSignal { signal, signalDisposable in
+						disposable += signalDisposable
+						signal.observe(Signal.Observer { event in
+							switch event {
+							case let .Next(value):
+								if let basename = value.value.lastPathComponent?.stringByDeletingPathExtension, fileUUID = NSUUID(UUIDString: basename) {
+									if remainingUUIDs.remove(fileUUID) != nil {
+										observer.put(event)
+										if remainingUUIDs.isEmpty {
+											sendCompleted(observer)
```

Comment:
Right, but the worry is that the dictionary is actually being captured by a `SignalProducer`, not just a `Signal`, and if that `SignalProducer` were to be started multiple times, the resulting signals would all misbehave. My belief is that, because the `SignalProducer` is immediately handed back to `flatMap`, the merge strategies only ever started the `SignalProducer` once and it's not accessible to anyone else. I just want confirmation that there's absolutely no way to have that `SignalProducer` object get started multiple times.